### PR TITLE
feat: add final structure for trezor wallet support

### PR DESCRIPTION
**Pull request by lucasgabrielgsp**
- [x] add final structure for trezor wallet support

### CODE REVIEW

File: packages/sysweb3-keyring/src/transactions/syscoin.ts [hunk 1]
```diff
@@ -795,13 +835,151 @@ export class SyscoinTransactions implements ISyscoinTransactions {
...
+        const value = new sys.utils.BN(amount * 1e8);
+
+        let outputs = [
+          {
+            address: receivingAddress,
+            value,
+          },
+        ];
+
+        const changeAddress = await this.trezor.getAddress({
```
Comment (claudiovb) [inline packages/sysweb3-keyring/src/transactions/syscoin.ts:932]: This should be fetched straight from the blockbook, we must only rely on trezor for fetching initially the account and signing the transaction

**Review by claudiovb — CHANGES_REQUESTED** [review]

File: packages/sysweb3-keyring/src/keyring-manager.ts [hunk 2]
```diff
@@ -673,62 +676,79 @@ export class KeyringManager implements IKeyringManager {
...
+
+    const isEVM = coin === 'eth';
+
+    const address = isEVM
+      ? xpub
+      : this.syscoinTransaction.getAddress(xpub, +index, false);
+
+    if (isEVM) {
+      const response = await this.trezorSigner.getPublicKey({
+        coin,
```
Comment (claudiovb) [inline packages/sysweb3-keyring/src/keyring-manager.ts:698]: Doesn't getAccountInfo provides already a xpub? Depending on how you configure the derivation path (for Ethereum we get the address so 5 parameters do make sense) but for Bitcoin and utxo networks where we're interested in the xpub we can stop into the third level of the derivation path, something like this should work:
```js
async function getXpubFromAccountInfo() {
  // Request the account info
  const result = await TrezorConnect.getAccountInfo({
    coin: 'Bitcoin', // Use the desired coin name
    path: "m/49'/0'/0'", // Use the desired derivation path
  });

  if (result.success) {
    // Extract the xpub from the response
    const xpub = result.payload.descriptor;

    console.log('xpub:', xpub);
    return xpub;
  } else {
    console.error('Error:', result.payload.error);
    return null;
  }
}

getXpubFromAccountInfo();
```

File: packages/sysweb3-keyring/src/transactions/ethereum.ts [hunk 3]
```diff
@@ -244,6 +263,56 @@ export class EthereumTransactions implements IEthereumTransactions {
...
+      maxFeePerGas: `${params.maxFeePerGas.toHexString()}`,
+      // @ts-ignore
+      maxPriorityFeePerGas: `${params.maxPriorityFeePerGas.toHexString()}`,
+      // @ts-ignore
+      value: `${params.value.toHexString()}`,
+      nonce: this.toBigNumber(transactionNonce)._hex,
+      chainId: activeNetwork.chainId,
+    };
+
+    if (activeAccountType === KeyringAccountType.Trezor) {
```
Comment (claudiovb) [inline packages/sysweb3-keyring/src/transactions/ethereum.ts:288]: This verification should be done before, we need to  split sendFormattedTransaction into two subfunctions, one for Imported and HDAccount. Other for Trezor.
That way we avoid doing not needed computation for trezor before the time. We can get the parameters that will be needed for both before routing to one or other function and passing them as arguments something like this:
```js
sendFormattedTransaction = async (params: SimpleTransactionRequest) => {
    const { decryptedPrivateKey } = this.getDecryptedPrivateKey();
    ... fetch needed parameters by both
    switch (activeAccountType):
    case 0: 
    function0( neededParams);
    break;
    case 1: 
    function1( neededParams);
    break;
    case ...
    default:
    
    ...
    
 
```

**Review by claudiovb — CHANGES_REQUESTED** [review]
Besides those changes there are some other more structural points on ClickUp check it up there as well.

File: packages/sysweb3-keyring/src/keyring-manager.ts (hunk 2 above)
Comment (lucasgabrielgsp) [reply packages/sysweb3-keyring/src/keyring-manager.ts:702]: For EVM networks, the method TrezorConnect.getAccountInfo returns in descriptor the address of account requested with derivation path. In this case, we need to use an independent method to get EVM public key instead getAccountInfo.

**Review by claudiovb — APPROVED** [review]

**Pull request merged**